libvast/src/format/zeek.cpp

```diff
@@ -232,35 +232,6 @@ const char* reader::name() const {
   return "zeek-reader";
 }
 
-void reader::patch(std::vector<data>& xs) {
-  auto protocol = port::unknown;
-  // Get the protocol from the proto field if available.
-  if (proto_field_) {
-    VAST_ASSERT(*proto_field_ < xs.size());
-    if (auto proto_string = caf::get_if<std::string>(&xs[*proto_field_])) {
-      auto p = parsers::port_type >> parsers::eoi;
-      if (!p(*proto_string, protocol))
-        VAST_DEBUG(this, "could not parse protocol", *proto_string);
-    }
-  }
-  // Or use a simple heuristic if not.
-  else {
-    if (type_.name() == "zeek.ftp" || type_.name() == "zeek.http"
-        || type_.name() == "zeek.irc" || type_.name() == "zeek.rdp"
-        || type_.name() == "zeek.smtp" || type_.name() == "zeek.ssh"
-        || type_.name() == "zeek.xmpp")
-      protocol = port::tcp;
-    else if (type_.name() == "zeek.dhcp" || type_.name() == "zeek.dns"
-             || type_.name() == "zeek.smnp")
-      protocol = port::udp;
-  }
-  // Assign the deduced proto to all port fields.
-  for (auto& pf : port_fields_) {
-    auto& p = caf::get<port>(xs[pf]);
-    p.type(protocol);
-  }
-}
```

Comment:
I wonder if removing this function has a big effect on the performance of the Zeek reader. The benchmarks should be much quicker now. Can't believe we've done it this way.

Reply:
Yeah, although I have no good intuition how these checks will effect the end-to-end performance. Only measurements will tell...